src/Microsoft.AspNet.Mvc.Core/ParameterBinding/ModelBindingHelper.cs

```diff
@@ -381,5 +387,54 @@ public static string CreatePropertyModelName(string prefix, string propertyName)
                 return prefix + "." + propertyName;
             }
         }
+
+        internal static void ClearModelStateDictionaryForModel(
+            [NotNull] Type modelType,
+            [NotNull] ModelStateDictionary modelstate,
+            [NotNull] IModelMetadataProvider metadataProvider,
+            string modelKey)
+        {
+            if (string.IsNullOrEmpty(modelKey))
```

Comment:
Would love to see a comment here about what these cases mean and why the null/empty prefix is so much more complicated.